plugin/http/propagation/propagation.go

```diff
@@ -0,0 +1,86 @@
+// Copyright 2018, OpenCensus Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package propagation contains HTTP propagators.
+package propagation
+
+import (
+	"encoding/hex"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"go.opencensus.io/trace"
+)
+
+const (
+	supportedVersion = 0
+	header           = "TraceContext"
+)
+
+// HTTPFormat implements the TraceContext trace propagation format.
+// See https://github.com/w3c/distributed-tracing for more information.
+type HTTPFormat struct{}
+
+// FromRequest extracts a TraceContext span context from incoming requests.
+func (f *HTTPFormat) FromRequest(req *http.Request) (sc trace.SpanContext, ok bool) {
+	h := req.Header.Get(header)
+	if h == "" {
+		return trace.SpanContext{}, false
+	}
+	sections := strings.Split(h, "-")
+	if len(sections) < 3 {
+		return trace.SpanContext{}, false
+	}
+
+	ver, err := hex.DecodeString(sections[0])
+	if err != nil {
+		return trace.SpanContext{}, false
+	}
+	if len(ver) == 0 || int(ver[0]) > supportedVersion {
+		return trace.SpanContext{}, false
+	}
+
+	tid, err := hex.DecodeString(sections[1])
```

Comment:
Perhaps we could use a helper like this to enforce a 16 byte TraceID
```go
func parseTraceID(strTraceID string) (sid []byte, err error) {
     // TraceID must be a 16 byte string
     if g, w := len(strTraceID), 32; g != w {
        return nil, fmt.Errorf("TraceID got length=%d want length=w", g, w)
     }
    sid, err = hex.Decode(strTraceID)
    if err != nil {
       return nil, err
    }
    if bytes.Equal(sid, all16BlankBytes[:]) {
       return nil, errors.New("expected non-all zero traceID")
    }
    return sid, nil
}
var all16BlankBytes = [16]byte
```

Reply:
Checking the length now. We don't need this expensive check but just make sure the length is 16 bytes for trace IDs and 8 bytes for span IDs.